helpers/calendar/all_analytics_helper.py

```diff
@@ -0,0 +1,107 @@
+import graphene
+import dateutil.parser
+from graphql import GraphQLError
+from utilities.utility import percentage_formater
+from helpers.calendar.analytics_helper import CommonAnalytics
+
+
+class Event(graphene.ObjectType):
+    duration_in_minutes = graphene.Int()
+
+
+class BookingsCount(graphene.ObjectType):
+    period = graphene.String()
+    bookings = graphene.Int()
+
+
+class AllAnalyticsHelper:
+
+    def bookings_count(self, unconverted_dates, events):
+        """
+        Get bookings count and period in a given room
+        """
+        start_date = unconverted_dates['start']
+        day_after_end_date = unconverted_dates['end']
+        bookings_count = []
+        start_dt = dateutil.parser.parse(start_date)
+        end_dt = dateutil.parser.parse(day_after_end_date)
+        number_of_days = (end_dt - start_dt).days
+
+        if number_of_days <= 30:
+            dates = CommonAnalytics.get_list_of_dates(
+                start_date, number_of_days)
+            for date in dates:
+                bookings = 0
+                string_date = dateutil.parser.parse(
+                    date[0]).strftime("%b %d %Y")
+                for event in events:
+                    if dateutil.parser.parse(event.start_time[:10]).strftime("%b %d %Y") == string_date: # noqa
+                        bookings += 1
+                output = BookingsCount(
+                    period=string_date,
+                    bookings=bookings)
+                bookings_count.append(output)
+        else:
+            dates = CommonAnalytics.get_list_of_month_dates(
+                start_date, start_dt, day_after_end_date, end_dt)
+            for date in dates:
+                string_month = dateutil.parser.parse(date[0]).strftime("%B")
+                bookings = 0
+                for event in events:
+                    if dateutil.parser.parse(event.start_time[:10]).strftime("%B") == string_month: # noqa
+                        bookings += 1
+                output = BookingsCount(
+                    period=string_month,
+                    bookings=bookings)
+                bookings_count.append(output)
+
+        return bookings_count
+
+    def get_all_analytics(self, query, start_date, end_date, location_id, unconverted_dates): # noqa
+        """
+        Get all room analytics
+        """
+        rooms = query.filter_by(state="active", location_id=location_id).all()
+        room_analytics = []
+        bookings = 0
+        for room in rooms:
+            cancellations = 0
+            checkins = 0
+            events = []
+            try:
+                # events_result - number of events in a given room
```

Comment:
Remove commented out codes that are not doing anything significant to this method

Reply:
Fixed